griptape/drivers/prompt/amazon_bedrock_prompt_driver.py

```diff
@@ -1,18 +1,21 @@
 from __future__ import annotations
 import json
 from typing import TYPE_CHECKING, Any
-import boto3
 from attr import define, field, Factory
 from griptape.artifacts import TextArtifact
+from griptape.utils import import_optional_dependency
 from .base_multi_model_prompt_driver import BaseMultiModelPromptDriver
 
 if TYPE_CHECKING:
     from griptape.utils import PromptStack
+    import boto3
```

Comment:
I'm not familiar with this pattern, so just want to air out my understanding of what's happening here. `boto3` is an expensive import so we don't want to actually import it in the case where this file gets imported but isn't actually used. Then, on [line 17](https://github.com/griptape-ai/griptape/pull/298/files#diff-9464a93e53508f56e9787cb4e4a6192711a13ea4a00573bdd151298242bb31e9R17), we import it regardless because we're now actually using code that depends on the import. Is that right?

Reply:
Code guarded by `TYPE_CHECKING` will only get run by a static type checker, not during actual runtime.

So we can import `boto3` for its  _type_ `boto3.Session` without actually importing the module.

Then, the actual module import is done at runtime lazily using `import_optional_dependency("boto3")`